src/org/jetbrains/plugins/scala/compiler/HydraCompilerSettings.scala

```diff
@@ -0,0 +1,89 @@
+package org.jetbrains.plugins.scala.compiler
+
+import java.io.File
+
+import com.intellij.openapi.components._
+import com.intellij.openapi.project.Project
+import org.jetbrains.plugins.scala.compiler.SourcePartitioner.Auto
+
+import scala.beans.BeanProperty
+
+/**
+  * @author Maris Alexandru
+  */
+@State(
+        name = "HydraSettings",
+        storages = Array(new Storage("hydra.xml"))
+)
+class HydraCompilerSettings(project: Project) extends PersistentStateComponent[HydraCompilerSettingsState] {
+
+  private val ProjectRoot: String = getProjectRootPath
+
+  var isHydraEnabled: Boolean = false
+
+  var hydraVersion: String = "0.9.5"
+
+  var noOfCores: String = Math.ceil(Runtime.getRuntime.availableProcessors()/2D).toInt.toString
+
+  var hydraStorePath: String = getDefaultHydraStorePath
+
+  var sourcePartitioner: String = Auto.value
+
+  override def getState: HydraCompilerSettingsState = {
+    val state = new HydraCompilerSettingsState()
+    state.hydraVersion = hydraVersion
+    state.noOfCores = noOfCores
+    state.isHydraEnabled = isHydraEnabled
+    state.hydraStorePath = hydraStorePath
+    state.sourcePartitioner = sourcePartitioner
+    state.projectRoot = ProjectRoot
+    state
+  }
+
+  override def loadState(state: HydraCompilerSettingsState): Unit = {
+    isHydraEnabled = state.isHydraEnabled
+    hydraVersion = state.hydraVersion
+    noOfCores = state.noOfCores
+    hydraStorePath = state.hydraStorePath
+    sourcePartitioner = state.sourcePartitioner
+  }
+
+  def getDefaultHydraStorePath: String = ProjectRoot + File.separator + ".hydra"
+
+  private def getProjectRootPath: String = project.getBaseDir.getPresentableUrl
+}
+
+object HydraCompilerSettings {
+  def getInstance(project: Project): HydraCompilerSettings = ServiceManager.getService(project, classOf[HydraCompilerSettings])
+}
+
+class HydraCompilerSettingsState {
+  @BeanProperty
+  var isHydraEnabled: Boolean = false
+
+  @BeanProperty
+  var hydraVersion: String = ""
+
+  @BeanProperty
+  var noOfCores: String = ""
+
+  @BeanProperty
+  var hydraStorePath: String = ""
+
+  @BeanProperty
+  var sourcePartitioner: String = ""
+
+  @BeanProperty
+  var projectRoot: String = ""
```

Comment:
Can we have these as Path or because of serialization/deserialization we need to keep using String? (wonder what they do in the Scala support)

Reply:
We need to keep this as String because of serialization/deserialization